### ISSUE TITLE
Missing required GCS remote state configuration project since Terragrunt 0.44.2

### ISSUE THREAD

**Issue by phgie** [gruntwork-io/terragrunt#2477]
Hey there, since updating to Terragrunt 0.44.2 we have problems with our GCS remote state configuration. 

```
remote_state {
  backend = "gcs"
  generate = {
    path      = "backend.tf"
    if_exists = "overwrite_terragrunt"
  }
  config = {
    bucket = "<tf-state-bucket>"
    prefix = "terragrunt/${path_relative_to_include()}"

    impersonate_service_account = "${local.impersonate_service_account}"
  }
}
```

```
ERRO[0000] 1 error occurred:
        * Missing required GCS remote state configuration project
```

Reverting to 0.44.1 fixes the error.

The debug logs show terragrunt searching for the bucket but nothing changed on the remote end or at the authentication level. The only difference is the terragrunt version.

```
DEBU[0000] Initializing remote state for the gcs backend  prefix=[<...>] 
DEBU[0000] Remote state GCS bucket <tf-state-bucket> does not exist. Attempting to create it  prefix=[<...>] 
```

Were there any breaking changes regarding the GCS remote state configuration? I can see that 0.44.4 introduces new authentication method, but the error appears for 0.44.2 AND 0.44.4 but not for 0.44.1

Anything I can try on my side?

**Comment by phgie** [comment]
Further testing reveals that removing the service account impersonation also fixes the problem.

**Comment by splichy** [comment]
It's probably caused by an update of the underlying dependency cloud.google.com/go/storage from v1.16.0 to v1.27.0 introduced by github.com/hashicorp/go-getter v1.6.2 => v1.7.0 #2454

**Comment by phgie** [comment]
Any chance on this getting resolved? Service Account Impersonation with remote state configuration is just broken at the moment. :-(

**Comment by denis256** [comment]
Fix released in https://github.com/gruntwork-io/terragrunt/releases/tag/v0.50.6

**Comment by yinzara** [comment]
This doesn't seem to be actually fixed. I'm getting this error with the current version today.

**Comment by erdia7** [comment]
@yinzara is correct. I am also still getting this error with the current version as well.

**Comment by yinzara** [comment]
Could we get this issue reopened?

**Comment by JamesVorder** [comment]
For anyone else landing here, the solution to this issue allowed me to impersonate a service account: https://github.com/gruntwork-io/terragrunt/issues/3730

TL;DR: You must specify a `project` and `location` in your `remote_state.config` in order to use `GOOGLE_APPLICATION_CREDENTIALS` for service acccount impersonation.
